### ISSUE TITLE
Cop idea: Replace nested hash syntax with `Hash#dig`

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12049]
## Actual behavior

I have the following code:

```ruby
# bad
hsh.try(:[], 'foo').try(:[], 'bar').try(:[], 'baz')
hsh&.fetch('foo', nil)&.fetch('bar', nil)&.fetch('baz', nil)
hsh&.dig('foo')&.dig('bar')&.dig('baz')
```

I run rubocop and don't get any offences:
```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop file.rb 
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Expected behavior

I expected to receive the code:
```ruby
hsh&.dig('foo', 'bar', 'baz')
```

## Rubocop

```bash
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.54.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.18.0
  - rubocop-factory_bot 2.23.1
  - rubocop-performance 1.18.0
  - rubocop-rails 2.20.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.22.0
  - rubocop-thread_safety 0.5.1
```

**Comment by KarlHeitmann** [comment]
Is there any cop similar to this, but to alert if you are using:

`config[:root_key][:nested_key]` 

and suggest you to do:

`config.dig(:root_key, :nested_key)`

I used rubocop version 1.59.0 and didn't raise an alarm for such code. I think it is easier to do the cop mentioned in my comment, and use that cop as starting point to do the cop suggested by @ydakuka . It is important to use another method that is not square brackets to reach nested keys in a hash. I used nested square brackets and forgot the root key may not exist, and later on stumbled with a problem on my app that could have been quickly solved with a rubocop alert.

I am willing to contribute with this cop and the one @ydakuka suggests, any tips and feedback about this comment is appreciated (or is somebody thinks I'm wrong and @ydakuka approach is easier than my suggestion).

**Comment by ydakuka** [comment]
@KarlHeitmann there is different behavior here.

```ruby
2.7.6 :001 > config = { root_key: { nested_key: 1 } }
 => {:root_key=>{:nested_key=>1}} 
2.7.6 :002 > config[:root_key][:nested_key]
 => 1 
2.7.6 :003 > config.dig(:root_key, :nested_key)
 => 1 
```
```ruby
2.7.6 :004 > config = {}
 => {} 
2.7.6 :005 > config[:root_key][:nested_key]
Traceback (most recent call last):
        4: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/bin/irb:23:in `<main>'
        3: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/bin/irb:23:in `load'
        2: from /home/ydakuka/.rvm/rubies/ruby-2.7.6/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        1: from (irb):5
NoMethodError (undefined method `[]' for nil:NilClass)
2.7.6 :006 > config.dig(:root_key, :nested_key)
 => nil
```

**Comment by KarlHeitmann** [comment]
Yes @ydakuka  there is a different behavior, thus it should not be auto corrected. My intention with that cop is -if it enabled- it will warn the developer: hey! you can get an error here if the hash does not contain the `:root_key` key! Would you rather use `dig` instead?

**Comment by ydakuka** [comment]
That's not within RuboCop's responsibilities. Additionally, I believe that most Ruby developers are familiar with `Hash#dig`.

**Comment by KarlHeitmann** [comment]
Sounds convincing... also, in my case rubocop will have to _infer_ the variable is a hash in order to raise the warning. Making suggestions about the type of variables you are using is probably the scope of a project like Solargraph.

**Comment by dvandersluis** [comment]
`Style/DigChain` already replaces `hsh&.dig('foo')&.dig('bar')&.dig('baz')` with `hsh&.dig('foo', 'bar', 'baz')`.

I opened #13952 to handle the other examples.
